core/arch/interp.c

```diff
@@ -2836,8 +2836,13 @@ client_process_bb(dcontext_t *dcontext, build_bb_t *bb)
 #    ifdef X86
         if (!d_r_is_avx512_code_in_use()) {
             if (ZMM_ENABLED()) {
-                if (instr_may_write_zmm_register(inst))
+                if (instr_may_write_zmm_register(inst)) {
+                    LOG(THREAD, LOG_INTERP, 2, "Detected AVX-512 code in use\n");
+                    SELF_UNPROTECT_DATASEC(DATASEC_RARELY_PROT);
```

Comment:
Since this is in the main interpreter loop, I intentionally had put it here in order to call it only once for both setters. I had added a comment at the setter's declaration to point out the function assumes that the caller has done that. Sure to move? We had talked about performance here before.

Reply:
Actually, since this is called only once, this is not perf critical. I'll move it.

Reply:
Done.